### ISSUE TITLE
Add ability to override nodeAffinity for Jiva replicas

### ISSUE THREAD

**Issue by ssytnikov18** [openebs/openebs#3226]
## Description
The latest openebs adds `nodeAffinity` section to the Jiva replica deployments.
By default it uses `key: kubernetes.io/hostname`.
It should be configurable in StorageClass Jiva.

## Context
Here is the example how it looks like:

```yaml
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/hostname
                operator: In
                values:
                - ip-10-80-0-135.ec2.internal
```
If the node `ip-10-80-0-135.ec2.internal` removes from the cluster the pod won't be rescheduled to another node until I fix deployment manually.

`nodeAffinity` section should configurable to let the people configure there some Node labels or even disable it if it's not needed.

## Possible Solution
It could be done in the same way as `Replica NodeSelector Policy` https://docs.openebs.io/docs/next/jivaguide.html#Replica-NodeSelector-Policy

**Comment by caiconkhicon** [comment]
@prateekpandey14 : May I ask when will this feature be implemented? We are really struggling with it.

**Comment by kmova** [comment]
@ssytnikov18 - just a couple of more queries:
- How many nodes do you have in the cluster?
- When the nodes go offline? can they be controlled to go out one at a time? 

If you have 3 nodes in the cluster that are running the jiva volumes and you have PDB or other rules setup to ensure only one node goes out at a time, there is a quick solution that we can provide.

**Comment by ssytnikov18** [comment]
@kmova 

> How many nodes do you have in the cluster?
we have 30 worker nodes.

> When the nodes go offline? can they be controlled to go out one at a time?
usually only one node at a time go offline.

we have a few Jiva PVCs for different applications.
ReplicaCount = 3

**Comment by ssytnikov18** [comment]
for now my solution is manual:
after cluster recreation or deploying new app with Jiva PVC I'm editing deployments for Jiva replicas to remove ** nodeAffinity** section.
After that the replicas automatically rescheduling to the new nodes if necessary.

I just want to have this configurable during new Jiva PVC deployment (in storage class for example)

**Comment by ssytnikov18** [comment]
Implemented in release [v2.4.0](https://github.com/openebs/openebs/releases/tag/v2.4.0)

**Comment by kmova** [comment]
@ssytnikov18 - would it be possible to share details on how OpenEBS is helping you here. https://github.com/openebs/openebs/issues/2719

It would really help with pushing the OpenEBS through incubation and also making sure the listed use-cases are covered in the Release E2e test pipelines.
